### PR TITLE
Design notes for pinned message

### LINKED ISSUE #16876: Design review: pinned message
- [x] 1. Remove identicoin ring on user avatar - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3709:473614&mode=design#511507832)  
- [x] 2. Small icon - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3709:473614&mode=design#511507482)  
- [x] 3. too heavy/dark font (**Android only**)- comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3709:473614&mode=design#511509265)  
- [x] 4. Icon is not aligned (**Android only**) - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3709:473614&mode=design#511509855)  
- [x] 5. Pressed state doesn't have rounded corners and correct margins - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3709:473614&mode=design#511508797)

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #16876
status: ready

### CODE REVIEW

File: src/status_im2/contexts/chat/composer/reply/style.cljs [hunk 1]
```diff
@@ -6,8 +6,9 @@
   [pin? in-chat-input?]
   {:flex-direction :row
    :height         (when-not pin? 24)
-   :margin-left    (if (and (not in-chat-input?) (not pin?)) 26 (if platform/android? 4 0))
+   :margin-left    (when (and (not in-chat-input?) (not pin?)) 26)
```
Comment (OmarBasem) [inline src/status_im2/contexts/chat/composer/reply/style.cljs:8]: Are you sure about this change?

Comment (ibrkhalil) [reply src/status_im2/contexts/chat/composer/reply/style.cljs:9]: Tested the flow thoroughly after changing it and was able to not reproduce the issue and the switcher wasn't visible inside composer anymore.

Comment (ibrkhalil) [reply src/status_im2/contexts/chat/composer/reply/style.cljs:8]: It matches the design better without it.

**Review by OmarBasem — APPROVED** [review]

File: src/quo2/components/avatars/icon_avatar.cljs [hunk 2]
```diff
@@ -26,5 +26,7 @@
               :justify-content  :center
               :align-items      :center}}
      [icons/icon icon
-      {:size  icon-size
-       :color icon-color}]]))
+      {:size            icon-size
+       :container-style {:width  20
```
Comment (flexsurfer) [inline src/quo2/components/avatars/icon_avatar.cljs]: could you elaborate on why :container-style is needed here?

File: src/status_im2/contexts/chat/composer/reply/view.cljs [hunk 3]
```diff
@@ -60,15 +60,23 @@
...
+  [{:keys [from contact-name current-public-key disable-identicon?]}]
+  (let [display-name     (first (rf/sub [:contacts/contact-two-names-by-identity from]))
+        contact          (rf/sub [:contacts/contact-by-address from])
+        photo-path       (when-not (empty? (:images contact)) (rf/sub [:chats/photo-path from]))
+        remove-identicon (process-identicon photo-path disable-identicon?)]
     [rn/view {:style style/reply-from}
      [quo/user-avatar
       {:full-name         display-name
-       :profile-picture   photo-path
+       :profile-picture   remove-identicon
```
Comment (flexsurfer) [inline src/status_im2/contexts/chat/composer/reply/view.cljs]: `remove-identicon` sounds like a function

Comment (flexsurfer) [reply src/status_im2/contexts/chat/composer/reply/view.cljs]: what is identicon btw ? why `disable-identicon?` is needed?

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: src/status_im2/contexts/chat/composer/reply/view.cljs (hunk 3 above)
Comment (ibrkhalil) [reply src/status_im2/contexts/chat/composer/reply/view.cljs]: To remove the colored border that's around the user's avatar (Identicon) as it doesn't match design

File: src/quo2/components/avatars/icon_avatar.cljs (hunk 2 above)
Comment (ibrkhalil) [reply src/quo2/components/avatars/icon_avatar.cljs]: To make the icon container match the design in terms of it's dimensions

File: src/status_im2/contexts/chat/composer/reply/view.cljs (hunk 3 above)
Comment (flexsurfer) [reply src/status_im2/contexts/chat/composer/reply/view.cljs]: i think its named `ring` in the code, identicon is a generated avatar

File: src/quo2/components/avatars/icon_avatar.cljs (hunk 2 above)
Comment (flexsurfer) [reply src/quo2/components/avatars/icon_avatar.cljs]: icon has a size prop, width shouldn't be used

File: src/status_im2/contexts/chat/composer/reply/view.cljs [hunk 4]
```diff
@@ -60,15 +60,23 @@
      :style               style/reply-deleted-message}
     (i18n/label :t/message-deleted)]])
 
+(defn- process-identicon
+  "Makes identicon optional"
+  [{:keys [uri] :as photo-path} disable-identicon?]
+  (if disable-identicon?
+    (assoc photo-path :uri (string/replace uri #"&addRing=1" ""))
```
Comment (flexsurfer) [inline src/status_im2/contexts/chat/composer/reply/view.cljs]: does "&addRing=1" mean no ring ?

File: src/status_im2/contexts/chat/composer/reply/view.cljs (hunk 3 above)
Comment (flexsurfer) [reply src/status_im2/contexts/chat/composer/reply/view.cljs]: so my suggestion `photo-path-without-ring`

File: src/status_im2/contexts/chat/composer/reply/view.cljs (hunk 4 above)
Comment (ibrkhalil) [reply src/status_im2/contexts/chat/composer/reply/view.cljs]: It means ring, Replacing it with an empty string means we won't get a ring back

File: src/status_im2/contexts/chat/composer/reply/view.cljs (hunk 3 above)
Comment (ibrkhalil) [reply src/status_im2/contexts/chat/composer/reply/view.cljs]: okay

File: src/quo2/components/avatars/icon_avatar.cljs (hunk 2 above)
Comment (ibrkhalil) [reply src/quo2/components/avatars/icon_avatar.cljs]: What if we wanted to configure that icon's container's dimensions?

Comment (flexsurfer) [reply src/quo2/components/avatars/icon_avatar.cljs]: margins ?

Comment (ibrkhalil) [reply src/quo2/components/avatars/icon_avatar.cljs]: Margins control spacing I meant the container's sizing.

Comment (flexsurfer) [reply src/quo2/components/avatars/icon_avatar.cljs]: i'm not sure i understand what is container's sizing, why?

Comment (ibrkhalil) [reply src/quo2/components/avatars/icon_avatar.cljs]: I mean the sizing of the container that's around the icon, It's parent container.

Comment (flexsurfer) [reply src/quo2/components/avatars/icon_avatar.cljs]: but why margin or padding can't be used ?

**Comment by ibrkhalil** [comment]
@yqrashawn Fixed issue 1 [here](https://github.com/status-im/status-mobile/pull/17011)

File: src/quo2/components/avatars/icon_avatar.cljs (hunk 2 above)
Comment (flexsurfer) [reply src/quo2/components/avatars/icon_avatar.cljs]:
```
[rn/image
       {:style
        (merge {:width  size
                :height size}

               (when (not no-color)
                 {:tint-color (if (and (string? color) (not (string/blank? color)))
                                color
                                (colors/theme-colors colors/neutral-100 colors/white theme))})

               container-style)
```
so width and height managed by size prop, overriding them by ` container-style` is wrong, different size should be used instead

Comment (ibrkhalil) [reply src/quo2/components/avatars/icon_avatar.cljs]: Thanks Andrey!
Will address this tomorrow

Comment (flexsurfer) [reply src/quo2/components/avatars/icon_avatar.cljs]: do we have a component? if so there should be an icon size prop, if sometimes designers want a bigger icon, i believe it's only for some specific cases, so currently default size works fine for most cases

Comment (ibrkhalil) [reply src/quo2/components/avatars/icon_avatar.cljs]: Will post side by side comparison between using what's already there and what's proposed.

Comment (ibrkhalil) [reply src/quo2/components/avatars/icon_avatar.cljs]: Fixed, I had to adjust icon-avatar, Thanks for your comments!

Comment (ibrkhalil) [reply src/quo2/components/avatars/icon_avatar.cljs]: Kindyl take another look

File: src/quo2/components/messages/system_message.cljs [hunk 5]
```diff
@@ -151,7 +152,8 @@
   [system-message-base
    {:icon      {:icon    :i/pin
                 :color   (or customization-color :primary)
-                :opacity 5}
+                :opacity 5
+                :size    20}
```
Comment (flexsurfer) [inline src/quo2/components/messages/system_message.cljs:156]: 20 is the default size no need to specify it

Comment (flexsurfer) [inline src/quo2/components/messages/system_message.cljs]: should it be medium? and probably it should be by default, so we don't need to specify it

**Review by flexsurfer — APPROVED** [review]

**Comment by qoqobolo** [comment]
Thanks for the fixes @ibrkhalil, great job!
@Francesca-G PR is ready for design review, thanks!

**Review by Francesca-G — COMMENTED** [review]
Here's the [design review](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=4966%3A121633&mode=design&t=tuTFJhP4jRsfHuyE-1) :)

**Comment by ibrkhalil** [comment]
[Reference to this icon change](https://discord.com/channels/624634427930312714/975775548007678033/1146283883281793065)

**Comment by ibrkhalil** [comment]
> Here's the [design review](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=4966%3A121633&mode=design&t=tuTFJhP4jRsfHuyE-1) :)

Can you please take another look @Francesca-G ?

**Review by Francesca-G — APPROVED** [review]

**Pull request merged**